build.zig

```diff
@@ -114,7 +114,15 @@ pub fn addCargoBuild(b: *std.Build, config: CargoConfig, args: anytype) std.Buil
         if (zig_target.os.tag == .windows) {
             target.abi = .gnu;
         }
-        const rust_target = @This().Target.fromZig(target) catch @panic("unable to convert target triple to Rust");
+        const host_is_alpine_linux = b.graph.host.result.os.tag == .linux and blk: {
+            std.fs.cwd().access("/etc/alpine-release", .{}) catch break :blk false;
```

Comment:
This line tries accessing a file on a local machine which is not available in case of cross compilation.
The API already provides `config.target`, is it not enough to handle this case?

Reply:
> is it not enough to handle this case?

It is, but then this detection would have to be implemented from scratch in every project using this library which I don't really want.

Reply:
Would you be happy with something like this?

```zig
pub const Vendor = union(enum) {
    alpine,
    apple,
    esp,
    fortanix,
    ibm,
    kmc,
    linux,
    nintendo,
    none,
    nvidia,
    openwrt,
    pc,
    sony,
    sun,
    unikraft,
    unknown,
    uwp,
    wrc,
    custom: []const u8,
};

pub const Target = struct {
    arch: Arch,
    vendor: Vendor,
    os: Os,
    env: Env,

    const Options = struct {
        vendor: ?Vendor = null,
    };

    pub fn fromZig(target: std.Target, options: Options) error{Unsupported}!Target {
        return .{
            .arch = try Arch.fromZig(target),
            .vendor = options.vendor orelse try Vendor.fromZig(target),
            .os = try Os.fromZig(target.os),
            .env = try Env.fromZig(target),
        };
    }
```

and then as the caller I can do (using my own detection):

```
const rust_target = Target.fromZig(target, .{ .vendor = .{ .custom = "alpine" } }) catch @panic("unable to convert target triple to Rust");
```

Reply:
Done :)